src/stamps.py

```diff
@@ -30,27 +32,43 @@ class InvalidStampToken(Exception):
     pass
 
 
-def generate_stamp_token(employee_id: int, place_id: int) -> str:
+def generate_stamp_token(user_id: int) -> str:
     return __enc_box.encrypt(
-        f"{employee_id}\n{place_id}\n{(datetime.utcnow() + timedelta(seconds=30)).isoformat()}"
+        f"{user_id}\n{(datetime.utcnow() + timedelta(seconds=30)).isoformat()}"
         f"".encode('utf-8')
     ).hex()
 
 
-def make_stamp(token: str, visitor_id: int) -> Stamp:
+async def make_stamp(session: AsyncSession, token: str, employee_id: int) -> Stamp:
+    '''
+    Make a stamp based on token scanned from user and employee_Did
+
+    excepts:
+        employee doesn't exist or 
+    '''
+    employee = await Employees.by_id(session, employee_id)
+    if employee is None:
+        logger.debug("Employee doesn't exist!")
+        raise InvalidStampToken()
+
+    place_id = employee.get("place_id")
+    if place_id is None:
+        logger.debug("Employee doesn't have an assigned place!")
+        raise InvalidStampToken()
+
     try:
         # Decrypt the data from the token
         token: List[str] = __dec_box.decrypt(bytes.fromhex(token)) \
             .decode('utf-8').split('\n')
 
         logger.debug("Token decoded : {token}")
         logger.debug(f"time now :  {datetime.utcnow()}")
-        logger.debug(f"token time: {datetime.fromisoformat(token[2])}")
+        logger.debug(f"token time: {datetime.fromisoformat(token[1])}")
```

Comment:
Safe with the current implementation but far from ideal to be before the length check.